R/epidist_utils.R

```diff
@@ -0,0 +1,605 @@
+#' vector-borne (i.e. is transmitted between humans through an intermediate
+#' vector).
+#' @param vector The name of the vector transmitting the vector-borne disease.
+#' This can be a common name, or a latin binomial name of a specific vector
+#' species. Both the common name and taxonomic name can be given with one given
+#' in parentheses. When a disease is not vector-borne NA should be given.
+#' @param extrinsic A boolean value defining whether the data entry is an
+#' extrinsic delay distribution, such as the extrinsic incubation period.
+#' This field is required because intrinsic and extrinsic delay distributions
+#' are stored as separate entries in the database and can be linked.
+#' When the disease is not vector-borne FALSE should be given.
+#' @param inference_method The type of inference used to fit the delay
+#' distribution to the data. Abbreviations of model fitting techniques can be
+#' specified as long as they are non-ambiguous. This field is only used to
+#' determine whether the uncertainty intervals possibly specified in the other
+#' fields are: confidence intervals (in the case of maximum likelihood), or
+#' credible intervals (in the case of bayesian inference). Uncertainty bounds
+#' for another types of inference methods, or if the inference method is
+#' unstated are assumed to be confidence intervals. When the inference method
+#' is unknown or a disease does not have a probability distribution NA can be
+#' given.
+#'
+#' @return named list
+#' @export
+#'
+#' @examples
+#' # it will automatically populate the fields with defaults if left empty
+#' create_epidist_metadata()
+#'
+#' # supplying each field
+#' create_epidist_metadata(
+#' sample_size = 10,
+#' region = "UK",
+#' vector_borne = TRUE,
+#' vector = "mosquito",
+#' extrinsic = FALSE,
+#' inference_method = "MLE"
+#' )
+create_epidist_metadata <- function(sample_size = NA_integer_,
+                                    region = NA_character_,
+                                    vector_borne = FALSE,
+                                    vector = NA_character_,
+                                    extrinsic = FALSE,
+                                    inference_method = NA_character_) {
+
+  # check input
+  checkmate::assert_number(
+    sample_size,
+    na.ok = TRUE,
+    lower = 0,
+    finite = TRUE,
+    null.ok = TRUE
+  )
+  checkmate::assert_character(region)
+  checkmate::assert_logical(vector_borne, len = 1)
+  checkmate::assert_character(vector)
+  checkmate::assert_logical(extrinsic, len = 1)
+  checkmate::assert_character(inference_method)
+
+  if (isFALSE(vector_borne) && !is.na(vector)) {
+    stop("A vector is given for a non-vector-borne disease please check input")
+  }
+
+  # return list of metadata
+  list(
+    sample_size = sample_size,
+    region = region,
+    vector_borne = vector_borne,
+    vector = vector,
+    extrinsic = extrinsic,
+    inference_method = inference_method
+  )
+}
```

Comment:
1. Function title needs fixing.
2. Suggest better explaining `extrinsic()` as only applicable for vector borne diseases - suggest moving details to Details, where it would be good to explain what an extrinsic delay distribution is.
3. Suggest optionally better handling of `region` argument as a named list of "country", "province", "location", and possibly "other" for extra information. This functionality might help future review/meta-analysis studies which might want to count studies in each region/country, or to map locations of study sites from which parameters are available. This is something initiatives are increasingly interested in, see e.g. https://www.viralemergence.org/blog/building-a-global-atlas-of-wildlife-disease-data.
4. Value field requires expansion - update the `@return`  field.
5. Examples field should be styled - run `styler::style_pkg()`.

Reply:
Updated documentation (title, details and return) for `create_epidist_metadata()` in e70ccd5. Added `create_epidist_region()`, which for now is not used, but is a nice idea and can be used in future to standardise the geographical setting of each study.